### PR TITLE
Total Alignment

### PR DESCRIPTION
**Pull request by davidnewhall**
I'm going to rebase this (or at least merge in master) after #59 is tested and merged.

This converts all of the api methods to use the new `Request` struct. Renames the `Req` struct to `PageReq` (sorry) just to reduce confusion. Adds the `bpPathName` constants convention to all files. Converts most usage of `strconv` to `fmt.Sprint` and removes the weird `Base10` constant. 

Updates every method to use the new conventions:
- `path.Join` for all URIs.
- All requests get a `req := starr.Request{}`
- `bpPathName` for all path base names.
- `var output NonPointer` or `var output []*Pointer` for all outputs.
- Use `bpPathName`  or `&req` in all errors. The `&` is important.
- Place methods and associated structs into their own files (`sonarr/queue.go`, `sonarr/type.go`).
- All exported methods and structs have a `// Comment.`

Notes:
- It's entirely possible I forgot to pass a `Body: &body` into a method or two. I caught one with a test.
- Due to copy/paste/human mistakes it's possible I added an id to a request that was not supposed to have one. Caught one of these with a test too.
- I'd like to make a test that greps for the wrong `bpPathName` in files. It'll probably be slow, but may catch more human mistakes.

### CODE REVIEW

**Comment by Fuochi** [comment]
@davidnewhall, I'll start making some branches from this one, but I'll wait to create PRs to when it's merged into master

**Comment by davidnewhall** [comment]
I'm already using this. It's pretty solid overall. I did find one bug I need to fix though. an int where it should be an int64.

File: http.go [hunk 1]
```diff
@@ -60,43 +65,49 @@ func (c *Config) req(ctx context.Context, method string, req Request) (*http.Res
...
-		return nil, parseNon200(httpReq, resp)
+		return nil, parseNon200(resp)
 	}
 
 	return resp, nil
 }
 
 // parseNon200 attempts to extract an error message from a non-200 response.
-func parseNon200(req *http.Request, resp *http.Response) error {
+func parseNon200(resp *http.Response) error {
```
Comment (davidnewhall) [inline http.go:75]: I'm not sure how well this code works. It needs a test, really...

**Comment by davidnewhall** [comment]
Damn this is big.

**Pull request merged**